Add a mono-mixin and overhaul the public API (#80)
Financial-Times/o-tabs: README.md

Tabs component for dividing content into meaningful sections.

> - [Usage](#usage)

Reviewer: We've been removing the top level "Usage" heading

<ul data-o-component="o-tabs" class="o-tabs o-tabs--buttontabs" role="tablist">
```

The __buttontabs__ style comes in two sizes that conform to the `o-buttons` sizes: medium and big. Medium is the default size and big can be applied by adding the class `o-tabs--big`.

### Options

* __Align right__: Add `o-tabs--alignright` to the root element.

### Mixin: `oTabs`

The `oTabs` mixin is used to output base styles as well as styles for several of the tab themes (primary and inverse). This output includes the `o-tabs` classes:

```scss
@include oTabs();
```

```css
.o-tabs {
    /* styles */
}
.o-tabs--buttontabs {
    /* styles */
}
.o-tabs--primary {
    /* styles */
}
/* etc. */
```

> If you wish to specify a subset of themes to output styles for, you can pass in options (see [themes](#themes) for available options):

Reviewer: ```suggestion
If you wish to specify more themes to output styles for, you can pass in options (see [themes](#themes) for available options):
```

<ul data-o-component="o-tabs" class="o-tabs o-tabs--buttontabs" role="tablist">
```

The __buttontabs__ style comes in two sizes that conform to the `o-buttons` sizes: medium and big. Medium is the default size and big can be applied by adding the class `o-tabs--big`.

#### Options

* __Align right__: Add `o-tabs--alignright` to the root element.

#### Silent mode

> As with all Origami components, o-tabs has a [silent mode](https://origami.ft.com/spec/v1/sass/#sass-silent-mode). To use its compiled CSS (rather than incorporating its mixins into your own Sass) set `$o-tabs-is-silent: false;` in your Sass before you import the o-tabs Sass:

Reviewer: We've been deleting silent mode documentation from READMEs in favour of the mono mixin. E.g.
https://github.com/Financial-Times/o-table#sass
https://github.com/Financial-Times/o-forms#sass

```sass
$o-tabs-is-silent: false;
@import 'o-tabs/main';
```

#### Mixin: `oTabs`

If using o-tabs in silent mode, you'll need to use the mixins outlined here to output styles.

The `oTabs` mixin is used to output base styles as well as styles for _all_ of the tab themes. This output includes the `o-tabs` classes:

```scss
@include oTabs();
```

```css
.o-tabs {
    /* styles */
}
.o-tabs--buttontabs {
    /* styles */
}
.o-tabs--primary {
    /* styles */
}
/* etc. */
```

If you wish to specify a subset of themes to output styles for, you can pass in options (see [themes](#themes) for available options):

```scss
@include oTabs($opts: (
    'themes': ('primary')
));
```

> #### Mixin: `oTabsButtonTabsAddTheme`

Reviewer: Does `oTabsButtonTabsAddTheme` need to be public if I can:
```
@include oTabs($opts: (
    'themes': ('primary', 'mono')
));
```

Author: Yep because `mono` isn't included by default, and the registry uses mono

Reviewer: But the above snippet would work? As a Sass user I can include non-default themes without `oTabsButtonTabsAddTheme`? Should `oComponentAdd` be saved for custom things?

Author: Wait I have a suggestion, easier to commit it and I can roll back if necessary

Author: See https://github.com/Financial-Times/o-tabs/pull/80/commits/4023e81efc2f5dfad1ed7c0d6cd0b31a53c90286
